cluster/gce/gci/configure-helper.sh

```diff
@@ -2349,6 +2349,17 @@ EOF
   fi
 }
 
+# Sets up the manifests of local dns cache agent for k8s addons.
+function setup-nodelocaldns-manifest {
+  setup-addon-manifests "addons" "dns/nodelocaldns"
+  local -r localdns_file="${dst_dir}/dns/nodelocaldns/nodelocaldns.yaml"
+  mv "${dst_dir}/dns/nodelocaldns/nodelocaldns.yaml.sed" "${localdns_file}"
+  # Replace the sed configurations with variable values.
+  sed -i -e "s/\\\$DNS_DOMAIN/${DNS_DOMAIN}/g" "${localdns_file}"
```

Comment:
Just a suggestion, I'd recommend sticking with the __DNS_DOMAIN__ format so you don't have to worry about escaping.